aspnetcore/client-side/dotnet-js-interop.md

```diff
@@ -235,7 +235,9 @@ public static class PrimitivesUsage
 
 ## JS Date
 
-Demonstrates JSImport of methods which have a JS Date object as its return or parameter.
+Demonstrates JSImport of methods which have a JS Date object as its return or parameter.  Dates are marshalled across interop by-value, meaning they are copied in much the same way as JS primitives.
+
+Be aware that JS Date objects do not contain timezone information and are internally represented in UTC.  To ensure a reliable conversion when marshalling to JavaScript, a .NET DateTime should be initialized with DateTimeKind.Utc or DateTimeKind.Local, avoiding DateTimeKind.Unspecified.  Consider a scenario where the local environment's timezone is EST. If you pass the DateTime `new DateTime(1968, 12, 21, 7, 51, 0, DateTimeKind.Local)` to JavaScript and it were later returned back to .NET, then it will have DateTimeKind.Utc with the time adjusted to 12:51 PM UTC.  Although the original DateTimeKind was changed, the time is adjusted correctly.
```

Comment:
I agree that converting to Utc makes things easy to understand conceptually. But the browser also has (only) local timezone and `Date.toString()` will use that to format the date. It will not use utc timezone for formatting even if you created the JS Date instance this way!

"should be initialized with DateTimeKind.Utc or DateTimeKind.Local, avoiding DateTimeKind.Unspecified." is probably too prescriptive. Some people know how timezones work and this would seem as functional limitation, which we don't have.

Maybe we could say something like ".NET DateTime initialized with DateTimeKind.Utc or DateTimeKind.Local makes it easier to reason about"

I guess explaining how timezones work in dotnet and in the browser and how that interacts is out of scope for this article.  All I wanted here is hint about timezones. 

only my 2c, I don't insist

Reply:
"I agree that converting to Utc makes things easy to understand conceptually. But the browser also has (only) local timezone and Date.toString() will use that to format the date. "

Right, I was more focused on how do you ensure it's still effectively an accurate time.  Similar to avoiding the pitfall with JSON serialization of DateTimes where dev assigns local time but doesn't initialize Kind.

How about this:

"Be aware that a [JS Date](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date) is time-zone agnostic.   A .NET DateTime will be adjusted relative to its DateTimeKind when marshalled, but time zone information will not be preserved.  Consider initializing DateTime's with a DateTimeKind.Utc or DateTimeKind.Local consistent with the value it represents."

We can drop the last sentence if it still seems too prescriptive.

Reply:
yes, thanks